### ISSUE TITLE
open `Installed Packages` blocks atom

### ISSUE THREAD

**Issue by despairblue** [atom/settings-view#551]
I have 174 packages installed and when opening the installed packages view I could actually make myself a coffee since it renders atom unusable for about a minute :smile: 

![screenshot from 2015-06-24 01-16-24](https://cloud.githubusercontent.com/assets/927609/8319549/4fc0c654-1a10-11e5-961e-429df22a59d1.png)
![screenshot from 2015-06-24 01-17-25](https://cloud.githubusercontent.com/assets/927609/8319550/4fc179a0-1a10-11e5-8955-8e027afba5f3.png)

Could also be related to [this](https://github.com/facebook/nuclide/issues/84), maybe

**Comment by Sterbic** [comment]
Have you tried to build Nuclide from source?

**Comment by despairblue** [comment]
Nope, should I uninstall all nuclide packages before attempting this?

**Comment by despairblue** [comment]
Or does it install that packages into some other path and open atom with another user profile? Not quite sure what difference building the packages should make.

**Comment by Sterbic** [comment]
Uninstall everything through the GUI or by `rm -r ~/.atom/packages/nuclide-*`, than try installing from source. I am not sure myself about the difference but it worked for me.

**Comment by despairblue** [comment]
That worked, the package list loads in about 2 seconds now. So it's save to assume that `nuclide-installer` is the culprit. Still wondering if it's something that needs to be fixed only there or as well here. :confused:

**Comment by Sterbic** [comment]
I don't think it's directly a problem with `nuclide-installer` as it only runs `apm install` over the 13 packages required by nuclide.

**Comment by mauriciord** [comment]
Same problem for me. Installing with `./scripts/dev/setup` is perfect now, but i want the `apm install nuclide-installer`.
Did you tried `sudo apm install nuclide-installer` ?

**Comment by bolinfest** [comment]
@thedaniel I suspect that Atom is making too many requests / UI updates when you have a large number of packages installed. When I run `process._getActiveRequests()`, I see a ton of stuff in flight.

I think a batch API of some sort would help.
